hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4212,6 +4218,48 @@ func (r *HostedClusterReconciler) serviceAccountSigningKeyBytes(ctx context.Cont
 	return privateKeyPEMBytes, publicKeyPEMBytes, nil
 }
 
+func (r *HostedClusterReconciler) reconcileKubevirtPlatformDefaultSettings(ctx context.Context, hc *hyperv1.HostedCluster) error {
+	// auto generate the basedomain by retrieving the default ingress *.apps dns.
+	baseDomainPassthrough := hc.Annotations[hyperv1.KubeVirtBaseDomainPassthroughAnnotation]
+	if baseDomainPassthrough == "true" && hc.Spec.DNS.BaseDomain == "" {
+		// get base domain from default ingress of management cluster
+		defaultIngressOperator := manifests.IngressDefaultIngressController()
+		err := r.Client.Get(ctx, client.ObjectKeyFromObject(defaultIngressOperator), defaultIngressOperator)
+		if err != nil {
+			return err
+		}
+
+		// For the KubeVirt platform, the basedomain can be autogenerated using
+		// the *.apps domain of the management/infra hosting cluster
+		// This makes the guest cluster's base domain a subdomain of the
+		// hypershift infra/mgmt cluster's base domain.
+		//
+		// Example:
+		//   Infra/Mgmt cluster's DNS
+		//     Base: example.com
+		//     Cluster: mgmt-cluster.example.com
+		//     Apps:    *.apps.mgmt-cluster.example.com
+		//   KubeVirt Guest cluster's DNS
+		//     Base: apps.mgmt-cluster.example.com
+		//     Cluster: guest.apps.mgmt-cluster.example.com
+		//     Apps: *.apps.guest.apps.mgmt-cluster.example.com
+		//
+		// This is possible using OCP wildcard routes
+		hc.Spec.DNS.BaseDomain = defaultIngressOperator.Status.Domain
```

Comment:
when no basedomain is set, we're detecting the mgmt cluster's default ingress domain and using that here.